Cesium3DTilesContent/src/I3dmToGltfConverter.cpp

```diff
@@ -0,0 +1,738 @@
+        featureTableBinaryData.data() + *parsedContent.scale);
+    for (unsigned i = 0; i < parsedContent.instancesLength; ++i) {
+      decodedInstances.scales[i] =
+          glm::vec3(rawScale[i], rawScale[i], rawScale[i]);
+    }
+  } else if (parsedContent.scaleNonUniform) {
+    const auto* rawScaleNonUniform = reinterpret_cast<const glm::vec3*>(
+        featureTableBinaryData.data() + *parsedContent.scaleNonUniform);
+    for (unsigned i = 0; i < parsedContent.instancesLength; ++i) {
+      decodedInstances.scales[i] = rawScaleNonUniform[i];
+    }
+  }
+  rebaseInstances(decodedInstances);
+  ByteResult byteResult;
+  if (header.gltfFormat == 0) {
+    // Need to recursively read the glTF content.
+    auto gltfUri = std::string(
+        reinterpret_cast<const char*>(gltfData.data()),
+        gltfData.size());
+    return get(subprocessor, gltfUri)
+        .thenImmediately(
+            [options, subprocessor](ByteResult&& byteResult)
+                -> CesiumAsync::Future<GltfConverterResult> {
+              if (byteResult.errorList.hasErrors()) {
+                GltfConverterResult errorResult;
+                errorResult.errors.merge(byteResult.errorList);
+                return subprocessor.asyncSystem.createResolvedFuture(
+                    std::move(errorResult));
+              }
+              return BinaryToGltfConverter::convert(
+                  byteResult.bytes,
+                  options,
+                  subprocessor);
+            })
+        .thenImmediately([subResult = std::move(subResult)](
+                             GltfConverterResult&& converterResult) mutable {
+          if (converterResult.errors.hasErrors()) {
+            subResult.gltfResult.errors.merge(converterResult.errors);
+          } else {
+            subResult.gltfResult = converterResult;
+          }
+          return subResult;
+        });
+  } else {
+    return BinaryToGltfConverter::convert(gltfData, options, subprocessor)
+        .thenImmediately([subResult = std::move(subResult)](
+                             GltfConverterResult&& converterResult) mutable {
+          if (converterResult.errors.hasErrors()) {
+            return subResult;
+          }
+          subResult.gltfResult = converterResult;
+          return subResult;
+        });
+  }
+}
+
+glm::dmat4
+composeInstanceTransform(size_t i, const DecodedInstances& decodedInstances) {
+  glm::dmat4 result(1.0);
+  if (!decodedInstances.positions.empty()) {
+    result = translate(result, glm::dvec3(decodedInstances.positions[i]));
+  }
+  if (!decodedInstances.rotations.empty()) {
+    result = result * toMat4(glm::dquat(decodedInstances.rotations[i]));
+  }
+  if (!decodedInstances.scales.empty()) {
+    result = scale(result, glm::dvec3(decodedInstances.scales[i]));
+  }
+  return result;
+}
+
+std::vector<glm::dmat4> meshGpuInstances(
+    GltfConverterResult& result,
+    const ExtensionExtMeshGpuInstancing& gpuExt) {
```

Comment:
This conveys intent more quickly for me?
```suggestion
std::vector<glm::dmat4> getExtMeshGpuInstances(
    const ExtensionExtMeshGpuInstancing& gpuExt,
    GltfConverterResult& result) {
```